asu-navigation-menu.php

```diff
@@ -8,38 +8,73 @@
 // Exit if accessed directly.
 defined( 'ABSPATH' ) || exit;
 
-$menu_name = 'primary';
-$menu_items = uds_wp_get_menu_formatted_array( $menu_name );
+// initialize variables.
+$menu_name   = 'primary';
+$menu_items  = uds_wp_get_menu_formatted_array( $menu_name );
+$current_uri = null;
+
+// attempt to retrieve the current page's URL.
+if ( isset( $_SERVER['REQUEST_URI'] ) ) {
+	$clean_uri   = sanitize_text_field( wp_unslash( $_SERVER['REQUEST_URI'] ) );
+	$current_uri = home_url( $clean_uri );
+}
 
 /**
  * Main navigation loop.
  */
 foreach ( $menu_items as $item ) :
-	// evaluate item for: single, children (one column), or grandchildren (multiple columns).
-	$item['depth'] = uds_wp_get_menu_depth( $item );
+	/**
+	 * The uds_wp_get_menu_dep() method does two things, returning the values in an array:
+	 *
+	 * 1) determines the depth of a particular top-level menu item
+	 * (single, children, grandchildren) for deciding how to render it
+	 *
+	 *  2) determines whether or not the current top-level menu item contains the currently
+	 * displayed page in order to highlight it if so
+	 */
+
+	// get an array for this menu item, and assign the depth to a local variable.
+	// early attempts to use switch() with $item_data array values did not work.
+	$item_data     = uds_wp_get_menu_depth( $item, $current_uri );
+	$item['depth'] = $item_data['depth'];
+
+	// set the CSS classname for this top-level item.
+	if ( $item_data['has_current'] ) {
+		$current_class = ' active';
```

Comment:
I would remove the space before ' active'

Reply:
Pushed a new commit that removes this space